version: '3.8'

services:
  dynamodb-init:
    build: .
    container_name: dynamodb-init
    environment:
      - AWS_REGION=us-east-1
      - AWS_ACCESS_KEY_ID=dummy
      - AWS_SECRET_ACCESS_KEY=dummy
      - IS_LOCAL=true
      - DYNAMODB_ENDPOINT=http://dynamodb-local:8000
      - ENV=dev
    depends_on:
      - dynamodb-local
    command: python data/dynamodbs.py
    volumes:
      - .:/app

  api:
    build: .
    container_name: chatbot-api
    ports:
      - "8001:8000"
    volumes:
      - .:/app
    environment:
      - AWS_REGION=us-east-1
      - AWS_ACCESS_KEY_ID=dummy
      - AWS_SECRET_ACCESS_KEY=dummy
      - COGNITO_USER_POOL_ID=dummy
      - COGNITO_CLIENT_ID=dummy
      - IS_LOCAL=true
      - DYNAMODB_ENDPOINT=http://dynamodb-local:8000
      - ENV=dev
    depends_on:
      - dynamodb-init

  dynamodb-local:
    image: amazon/dynamodb-local:latest
    container_name: dynamodb-local
    ports:
      - "8000:8000"
    command: "-jar DynamoDBLocal.jar -sharedDb"
    volumes:
      - dynamodb-data:/home/dynamodblocal/data
    working_dir: /home/dynamodblocal

volumes:
  dynamodb-data:
    driver: local 